### DISCUSSION TITLE
ability to import STL / export STL files

### DISCUSSION THREAD

**Discussion by bman2000** [sn4k3/UVtools#500] [Ideas]
Great program.

a great feature would be the ability to import an STL file so that tools can be used to solidify the model and then export to a slicer. The sliced file can then be imported back into UVtools for checking and any necessary adjustments. 

Thank you

**Comment by tslater2006** [comment]
I think processing of STL files directly in UVTools is outside the scope of the program but obviously @sn4k3 can weigh in, we do have the ability to "voxelize" a sliced file back to STL though I don't think its used in any practical use cases, more for the novelty of it.

**Comment by sn4k3** [comment]
UVtools have no 3D/mesh capabilities, that require a whole new program and it's not an easy task.
You need to look for alternatives that work with meshes, such as [Autodesk Meshmixer](https://www.meshmixer.com) even then you may find hard to do what you want clean and easy.

**Reply by jimb15122** [reply]
I really really could use the ability to open an stl file directly, especially if invoked from command line.  of course, all supports, etc would need to be embedded.  the only additional piece of info would be the desired layer height, and optionaly xyz scaling ratios.  if you had this feature, you probably could drop support for many of the import file types, and start applying pressure to the industry to stop creating new formats everytime an ant farts.
